codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/EnumGenerator.kt

```diff
@@ -129,23 +129,23 @@ open class EnumGenerator(
         meta.render(writer)
         writer.write("struct $enumName(String);")
         writer.rustBlock("impl $enumName") {
-            rust("/// Returns the `&str` value of the enum member.")
-            rustBlock("pub fn as_str(&self) -> &str") {
-                write("&self.0")
+            writer.rust("/// Returns the `&str` value of the enum member.")
```

Comment:
Why has `writer` been added here? And in general within this file?

Reply:
More accidents. I fixed this up and converted some of the `write` calls to `rust` and `docs` calls